### PR TITLE
osd: remove redundant code

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
retest this please.

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -9054,28 +9054,21 @@ void OSDService::adjust_pg_priorities(const vector<PGRef>& pgs, int newflags)
     return;
   }
   set<spg_t> did;
-  if (newflags & OFR_BACKFILL) {
+  bool is_force;
+  if (newflags)
```
Comment (tchaikov) [inline src/osd/OSD.cc]: please keep the curly brackets around the `for` loop.

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -9054,28 +9054,21 @@ void OSDService::adjust_pg_priorities(const vector<PGRef>& pgs, int newflags)
...
-    dout(10) << __func__ << " " << ((newflags & OFR_CANCEL) ? "cleared" : "set")
-	     << " force_" << ((newflags & OFR_BACKFILL) ? "backfill" : "recovery")
-	     << " on no pgs" << dendl;
-  } else {
-    dout(10) << __func__ << " " << ((newflags & OFR_CANCEL) ? "cleared" : "set")
-	     << " force_" << ((newflags & OFR_BACKFILL) ? "backfill" : "recovery")
-	     << " on " << did << dendl;
-  }
+      if (OFR_BACKFILL)
+        is_force = pg->set_force_backfill(!(newflags & OFR_CANCEL));
```
Comment (tchaikov) [inline src/osd/OSD.cc]: it helps with the readability to repeat the `did.insert(pg->pg_id)`.

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
@tchaikov like now? more diet, more readability.

File: src/osd/OSD.cc [hunk 3]
```diff
@@ -9054,28 +9054,22 @@ void OSDService::adjust_pg_priorities(const vector<PGRef>& pgs, int newflags)
...
     for (auto& pg : pgs) {
-      if (pg->set_force_backfill(!(newflags & OFR_CANCEL))) {
-	did.insert(pg->pg_id);
-      }
-    }
-  } else if (newflags & OFR_RECOVERY) {
-    for (auto& pg : pgs) {
-      if (pg->set_force_recovery(!(newflags & OFR_CANCEL))) {
-	did.insert(pg->pg_id);
+      if (OFR_BACKFILL || OFR_RECOVERY) {
```
Comment (tchaikov) [inline src/osd/OSD.cc]: this is wrong.

File: src/osd/OSD.cc [hunk 4]
```diff
@@ -9054,28 +9054,22 @@ void OSDService::adjust_pg_priorities(const vector<PGRef>& pgs, int newflags)
...
-	     << " on no pgs" << dendl;
-  } else {
-    dout(10) << __func__ << " " << ((newflags & OFR_CANCEL) ? "cleared" : "set")
-	     << " force_" << ((newflags & OFR_BACKFILL) ? "backfill" : "recovery")
-	     << " on " << did << dendl;
-  }
+
+  dout(10) << __func__ << " "
+    << ((is_cancel) ? "cleared" : "set") << " "
+    << "force_" << ((newflags & OFR_BACKFILL) ? "backfill" : "recovery")
```
Comment (tchaikov) [inline src/osd/OSD.cc]: nit, probably just put
```c++
<< (newflags & OFR_BACKFILL) ? "force_backfill" : "_force_recovery")
```

File: src/osd/OSD.cc [hunk 5]
```diff
@@ -9054,28 +9054,22 @@ void OSDService::adjust_pg_priorities(const vector<PGRef>& pgs, int newflags)
...
-	     << " force_" << ((newflags & OFR_BACKFILL) ? "backfill" : "recovery")
-	     << " on no pgs" << dendl;
-  } else {
-    dout(10) << __func__ << " " << ((newflags & OFR_CANCEL) ? "cleared" : "set")
-	     << " force_" << ((newflags & OFR_BACKFILL) ? "backfill" : "recovery")
-	     << " on " << did << dendl;
-  }
+
+  dout(10) << __func__ << " "
+    << ((is_cancel) ? "cleared" : "set") << " "
```
Comment (tchaikov) [inline src/osd/OSD.cc]: please remove the parentheses around `is_cancel`

**Comment by shinobu-x** [comment]
@tchaikov what about now?

**Pull request closed without merging**